docs/preview/features/security/auth/certificate.md

```diff
@@ -56,18 +56,33 @@ public class Startup
 {
     public void ConfigureServices(IServiceCollections services)
     {
-        services.AddScoped<ICachedSecretProvider(serviceProvider => new MyCachedSecretProvider());
-
-        var certificateAuthenticationConfig = 
-            new CertificateAuthenticationConfigBuilder()
-                .WithIssuer(X509ValidationLocation.SecretProvider, "key-to-certificate-issuer-name")
-                .Build();
-
-        services.AddScoped<CertificateAuthenticationValidator>(
-            serviceProvider => new CertificateAuthenticationValidator(certificateAuthenticationConfig));
-
-        services.AddMvc(
-            options => options.Filters.Add(new CertificateAuthenticationFilter()));
+        // Recommended to use the Arcus secret store instead.
+        // See https://security.arcus-azure.net/features/secret-store/ for more information.
+        services.AddScoped<ICachedSecretProvider>(serviceProvider => new MyCachedSecretProvider());
+
+        var certificateValidator =
+            new CertificateAuthenticationValidator(
+                new CertificateAuthenticationConfigBuilder()
+                    .WithIssuer(X509ValidationLocation.SecretProvider, "key-to-certificate-issuer-name")
+                    .Build());
+
+        services.AddSingleton(certificateValidator);
+
+        services.AddMvc(mvcOptions => 
+        {
+            // Adds certificate authentication to the request pipeline.
+            mvcOptions.Filters.Add(new CertificateAuthenticationFilter());
+
+            // Additional consumer-configurable options to change the behavior of the authentication filter.
+            var authOptions = new CertificateAuthenticationOptions
+            {
+                // Adds certificate authentication to the request pipeline with emitting security events during the authorization of the request.
+                // (default: `false`)
+                EmitSecurityEvents = true;
+            };
+
+            mvcOptions.Filters.Add(new CertificateAuthenticationFilter(authOptions));
```

Comment:
This makes me think whether we should add an additional extension on the `IFilterCollection` that lets the consumer configure the options inside a lambda.

Reply:
Added these extensions. Question though: these are (like the JWT extension) in the `Microsoft.AspNetCore.Mvc.Filters` namespace because the `FilterCollection` is in that namespace; but this is probably (always?) configured during startup with the `IServiceCollection` services. So, if a consumer uses `services.AddMvc(options => options.Filters.[our extension])` it will now show.
Should we move these extensions then to the `Microsoft.Extensions.DependencyInjection`, which is the same namespace as the `IServiceCollection`?
We can for the JWT extension create a new extension set and obsolete the one we have now if we go through with this.

Reply:
I think the current one is best since you'll do it on the filters directly, no?

Reply:
Ok, let's leave it then!